sdk/turing/_base_types.py

```diff
@@ -23,6 +23,11 @@ def to_dict(self):
 
         return dict(attribs)
 
+    def __eq__(self, other):
```

Comment:
> Since this has been implemented, do we still need the tests to use obj1.to_dict() == obj2.to_dict() ?

In `ensembler_test.py` the comparison happens between a wrapper class entity (`turing.PyfuncEnsembler`) and generated class entity (`turing.generated.models.PyfuncEnsembler`) hence these objects are not equal. The other way of dealing with it would be `assert obj1 == ClassName.from_open_api(obj2)`, is it less confusing? 

> And is comparing only the public fields

Yes, I'd say that it's sufficient for now. And if we care about the private fields, then they should be made public